tests/cbtest/gtest/test_uzfs.cc

```diff
@@ -410,6 +410,129 @@ uzfs_mock_rebuild_scanner_rebuild_comp(void *arg)
 	zk_thread_exit();
 }
 
+void
+verify_rebuild_io(int rebuild_fd, zvol_info_t *zinfo)
+{
+	int rc;
+	char *write_buf, *read_buf, *send_buf;
+	char *cbuf;
+	int clen;
+	int len = 4096;
+	metadata_desc_t *md;
+	time_t now;
+	uint64_t cnt;
+	zvol_io_hdr_t rebuild_hdr = { 0 };
+	struct zvol_io_rw_hdr *rw_hdr;
+
+	now = time(0);
+	cbuf = ctime(&now);
+	clen = strlen(cbuf) - 1;
+
+	rebuild_hdr.opcode = ZVOL_OPCODE_READ;
+	rebuild_hdr.flags = ZVOL_OP_FLAG_REBUILD;
+	rebuild_hdr.status = ZVOL_OP_STATUS_OK;
+	rebuild_hdr.offset = 4096;
+	rebuild_hdr.len = len + sizeof (struct zvol_io_rw_hdr);
+
+	rc = uzfs_zvol_socket_write(rebuild_fd, (char *)&rebuild_hdr, sizeof(rebuild_hdr));
+        EXPECT_NE(rc, -1);
+
+	send_buf = (char *)malloc(rebuild_hdr.len);
+	write_buf = send_buf + sizeof (struct zvol_io_rw_hdr);
+
+	for (int i = 0; i < len; i += clen) {
+		memcpy(write_buf + i, cbuf, clen);
+	}
+
+        cnt = zinfo->write_req_received_cnt;
+
+        rw_hdr = (struct zvol_io_rw_hdr *)send_buf;
+        rw_hdr->io_num = rand();
+        rw_hdr->len = len;
+
+        rc = uzfs_zvol_socket_write(rebuild_fd, (char *)send_buf, rebuild_hdr.len);
+        EXPECT_NE(rc, -1);
+
+        /* check for write cnt */
+        while (1) {
+                if ((zinfo->write_req_received_cnt != (cnt + 1)) &&
+                    (zinfo->write_req_received_cnt != (cnt + 2)))
+                        sleep(1);
+                else
+                        break;
+        }
+
+	/* Verify if rebuilding replica has written data to correct dataset */
+	read_buf = (char *)malloc(rw_hdr->len);
+	uzfs_read_data(zinfo->main_zv, read_buf, rebuild_hdr.offset, rw_hdr->len, &md);
+	EXPECT_EQ(NULL, md->next);
+	EXPECT_EQ(md->metadata.io_num, rw_hdr->io_num);
+	EXPECT_EQ(memcmp(write_buf, read_buf, sizeof (read_buf)), 0);
```

Comment:
sizeof will return only 8 here.. if I'm not wrong, intention here is to compare entire rw_hdr->len

Reply:
yes.